fleet/gateway.tf

```diff
@@ -48,6 +48,12 @@ resource "aws_lb_target_group" "gateway" {
   protocol = "HTTP"
   vpc_id   = "${var.vpc_id}"
 
+  stickiness {
+    enabled = "${var.lb_stickiness_enabled}"
+    type = "lb_cookie"
+    cookie_duration = "${var.lb_stickiness_cookie_duration}"
```

Comment:
From https://www.terraform.io/docs/providers/aws/r/lb_target_group.html
> cookie_duration - (Optional) The time period, in seconds, during which requests from a client should be routed to the same target. After this time period expires, the load balancer-generated cookie is considered stale. The range is 1 second to 1 week (604800 seconds). The default value is 1 day (86400 seconds).

Reply:
> To be able to use websockets, sticky load balancer sessions needs to be enabled.

What is behaviour for connection open for more than this time, with multiple nodes behind the load balancer (or same node being destroyed for failure, and re-deployed)? What is user experience? Is test planned exercising this time limit?

Reply:
>You enable sticky sessions at the target group level. You can also set the duration for the stickiness of the load balancer-generated cookie, in seconds. The duration is set with each request. Therefore, if the client sends a request before each duration period expires, the sticky session continues.

This means that if a WS session is longer than the cookie session it will be invalid on the next request and the target instance will rotate. Unfortunately this is the only supported way of stickiness currently supported by AWS.

The cookie is set before protocol upgrade is initiated and is not used afterwards, so it may expire before the connection is closed. However it will not drop the connection, just next reconnect attempt will probably redirect to a different node.

To support longer sessions we need to make some additional infrastructure or mechanism for keep alive (like reconnecting once a day). We can also implement some WS orchestration service but it will require changes on application (node) level.

We discussed a test suite, but it would require to setup an additional network with stickiness enabled to be able to test different scenarios. This will at least slow down the integration tests.

Reply:
> Therefore, if the client sends a request before each duration period expires, the sticky session continues.

@lucafavatella this statement with a let's say a day or two during should be perfectly fine